src/kakarot/instructions/system_operations.cairo

```diff
@@ -451,22 +395,11 @@ namespace CallHelper {
             self=ctx.memory, element_len=args_size, element=calldata, offset=args_offset
         );
 
-        // TODO: account for value_to_empty_account_cost in dynamic gas
```

Comment:
we delete all this because we'll fix the gas later?

Reply:
this is causing a lot of `range_check_ptr` error because of undeflow in 

```
let remaining_gas = ctx.gas_limit - ctx.gas_used
```